### ISSUE TITLE
Rescue errors from Addressable?

### ISSUE THREAD

**Issue by dentarg** [sferik/twitter#742]
Let's start with `IDN::Idna::IdnaError`

I get this error when using Addressable with [`idn-ruby`](https://github.com/sporkmonger/addressable/blob/master/lib/addressable/idna.rb#L20)

``` ruby
[37] pry(main)> client.status("664629330899836929").user.website
IDN::Idna::IdnaError: Output would be too large or too small (5)
from /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:38:in `toASCII'
```

``` ruby
[38] pry(main)> wtf??
Exception: IDN::Idna::IdnaError: Output would be too large or too small (5)
--
 0: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:38:in `toASCII'
 1: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:38:in `block in to_ascii'
 2: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:36:in `map'
 3: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:36:in `to_ascii'
 4: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/uri.rb:1071:in `normalized_host'
 5: /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/uri.rb:822:in `freeze'
 6: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable.rb:17:in `block in <module:Memoizable>'
 7: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/method_builder.rb:117:in `call'
 8: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/method_builder.rb:117:in `block (3 levels) in create_memoized_method'
 9: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/memory.rb:63:in `block (3 levels) in fetch'
10: /Users/dentarg/.gem/ruby/2.2.2/gems/thread_safe-0.3.5/lib/thread_safe/cache.rb:58:in `fetch'
11: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/memory.rb:62:in `block (2 levels) in fetch'
12: /Users/dentarg/.rubies/ruby-2.2.2/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
13: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/memory.rb:61:in `block in fetch'
14: /Users/dentarg/.gem/ruby/2.2.2/gems/thread_safe-0.3.5/lib/thread_safe/cache.rb:58:in `fetch'
15: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/memory.rb:60:in `fetch'
16: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/method_builder.rb:116:in `block (2 levels) in create_memoized_method'
17: /Users/dentarg/.gem/ruby/2.2.2/gems/twitter-5.15.0/lib/twitter/user.rb:92:in `website'
18: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/method_builder.rb:117:in `call'
19: /Users/dentarg/.gem/ruby/2.2.2/gems/memoizable-0.4.2/lib/memoizable/method_builder.rb:117:in `block (3 levels) in create_memoized_method'
```

[gist with output from `client.status("664629330899836929").attrs`](https://gist.github.com/dentarg/524c3f26e51aad0c3191#file-664629330899836929-attrs-rb)

I figured out why this happens, `freeze` in Addressable calls `normalized_host`

``` ruby
[58] pry(main)> Addressable::URI.parse("http://AcinusFallumTrompetumNullunCreditumVisumEstAtCuadLongumEtCefallumEst.com").normalized_host
IDN::Idna::IdnaError: Output would be too large or too small (5)
from /Users/dentarg/.gem/ruby/2.2.2/gems/addressable-2.3.8/lib/addressable/idna/native.rb:38:in `toASCII'
```

I have also seen another issue from Addressable in the very same area of the twitter gem (`user.website`). That is `Addressable::URI::InvalidURIError`.

Here is a stack trace (from [Sentry](https://www.getsentry.com/))

```
Addressable::URI::InvalidURIError: Absolute URI missing hierarchical segment: 'http://'
  from doe-provider.rb:5:in `<main>'
  from raven/base.rb:102:in `capture'
  from doe-provider.rb:21:in `block in <main>'
  from /app/lib/twitter_stream.rb:25:in `find_urls_by'
  from twitter/streaming/client.rb:37:in `filter'
  from twitter/streaming/client.rb:119:in `request'
  from twitter/streaming/connection.rb:22:in `stream'
  from twitter/streaming/response.rb:16:in `<<'
  from twitter/streaming/response.rb:16:in `<<'
  from twitter/streaming/response.rb:25:in `on_body'
  from twitter/streaming/response.rb:25:in `each'
  from twitter/streaming/response.rb:27:in `block in on_body'
  from twitter/streaming/response.rb:27:in `call'
  from twitter/streaming/client.rb:116:in `block in request'
  from /app/lib/twitter_stream.rb:35:in `block in find_urls_by'
  from /app/lib/twitter_stream.rb:72:in `extract_profile_website_urls'
  from memoizable/method_builder.rb:116:in `block (2 levels) in create_memoized_method'
  from memoizable/memory.rb:60:in `fetch'
  from thread_safe/cache.rb:58:in `fetch'
  from memoizable/memory.rb:61:in `block in fetch'
  from monitor.rb:211:in `mon_synchronize'
  from memoizable/memory.rb:62:in `block (2 levels) in fetch'
  from thread_safe/cache.rb:58:in `fetch'
  from memoizable/memory.rb:63:in `block (3 levels) in fetch'
  from memoizable/method_builder.rb:117:in `block (3 levels) in create_memoized_method'
  from memoizable/method_builder.rb:117:in `call'
  from twitter/user.rb:94:in `website'
  from addressable/uri.rb:138:in `parse'
  from addressable/uri.rb:138:in `new'
  from addressable/uri.rb:795:in `initialize'
  from addressable/uri.rb:2250:in `defer_validation'
  from addressable/uri.rb:2305:in `validate'
```

Unfortunately I haven't been able to reproduce `Addressable::URI::InvalidURIError` locally. Maybe the Twitter profiles it happened for have changed by now. So I don't know what data triggers it. But I saw it before starting to use Addressable with `idn-ruby`.

I have `object.attrs`, object being `Twitter::Tweet` as [extra context from Sentry](https://gist.github.com/dentarg/524c3f26e51aad0c3191#file-sentry-extra-context-invalid-uri-error-txt) (sorry about the formatting, copy-paste from Sentry does not include line breaks, and it's not real JSON that can be pretty formatted).

I'm using twitter 5.15.0 and Ruby 2.2.2. (I can see `IDN::Idna::IdnaError` on master (dd6457d) too.)

Should the twitter gem do something about these exceptions or should I be prepared to take care of this in my application? (I'm currently doing that)
